Tom/Classes/TomView.swift

```diff
@@ -0,0 +1,118 @@
+//
+//  TomView.swift
+//  Tom
+//
+//  Created by Yusuf Demirci on 30.03.21.
+//
+
+final public class TomView: UIView {
+
+    // MARK: Properties
+
+    private let stackView: UIStackView = {
+        let view: UIStackView = .init()
+        view.distribution = .fillEqually
+        return view
+    }()
+
+    private var lineViews: [LineView] = []
+    private var isAnimationEnabled: Bool = false
+
+    private let configuration: Configuration
+
+    public init(configuration: Configuration, frame: CGRect = .init()) {
+        self.configuration = configuration
+
+        super.init(frame: frame)
+
+        setup()
+    }
+
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+}
+
+// MARK: - Publics
+
+public extension TomView {
+
+    func start() {
+        isAnimationEnabled = true
+
+        lineViews.forEach { [weak self] lineView in
+            self?.animate(lineView: lineView)
```

Comment:
Can we unwrap the optional "self" with "guard" statement here? 

`guard let self = self else { return }`

Reply:
According to [this article](https://medium.com/flawless-app-stories/the-nested-closure-trap-356a0145b6d) we shouldn't use `guard let self` here because the animate function has tasks that won't work on runtime(delayed).